### PR TITLE
Hotfix/gson builder fix

### PR DESCRIPTION
**Pull request by agduncan94**
Fixes GSON builder so that it properly deserializes CWL v1.0 documents.

### CODE REVIEW

**Comment by agduncan94** [comment]
Temporarily closing this because I think I can move similar code into a function.

File: cwlavro-tools/src/main/java/io/cwl/avro/CWL.java [hunk 1]
```diff
@@ -145,42 +146,114 @@ public static Object getStub(final Object type, final String value) {
     /**
      * @return a gson instance that can properly convert CWL tools into a typesafe Java object
      */
-    public static Gson getTypeSafeCWLToolDocument() {
-        final Type hintType = new TypeToken<List<Any>>() {}.getType();
+    public static Gson getTypeSafeCWLToolDocument() throws GsonBuildException {
+        final java.lang.reflect.Type hintType = new TypeToken<List<Any>>() {}.getType();
```
Comment (denis-yuen) [inline cwlavro-tools/src/main/java/io/cwl/avro/CWL.java]: Is there a reason this switches from Type to java.lang.reflect.Type without the import?

File: cwlavro-tools/src/main/java/io/cwl/avro/CWL.java [hunk 2]
```diff
@@ -145,42 +146,114 @@ public static Object getStub(final Object type, final String value) {
...
+        return new GsonBuilder().registerTypeAdapter(CharSequence.class, (JsonDeserializer<CharSequence>) (json, typeOfT, context) -> json.getAsString())
+                .registerTypeAdapter(hintType, (JsonDeserializer) (json, typeOfT, context) -> {
+                    Collection<Object> hints = new ArrayList<>();
+                    for (final JsonElement jsonElement : json.getAsJsonArray()) {
+                        final Object o = sequenceSafeGson.fromJson(jsonElement, Object.class);
+                        hints.add(o);
+                    }
+                    return hints;
+
+                }).registerTypeAdapter(commandInputParameterType, (JsonDeserializer) (json, typeOfT, context) -> {
```
Comment (denis-yuen) [inline cwlavro-tools/src/main/java/io/cwl/avro/CWL.java]: Nice

File: cwlavro-tools/src/main/java/io/cwl/avro/CWL.java [hunk 3]
```diff
@@ -145,42 +146,114 @@ public static Object getStub(final Object type, final String value) {
...
+                            setType.invoke(item, itemObj.getValue());
+                        } else {
+                            setSource.invoke(item, itemObj.getValue());
+                        }
+                    }
+                    setId.invoke(item, itemObj.getKey());
+                    objectCollection.add(item);
+                }
+            } catch (InstantiationException ex) {
+                ex.printStackTrace();
```
Comment (denis-yuen) [inline cwlavro-tools/src/main/java/io/cwl/avro/CWL.java]: Get rid of these stacktraces. Replace it with a logging command.

File: cwlavro-tools/src/main/java/io/cwl/avro/CWL.java [hunk 4]
```diff
@@ -145,42 +146,114 @@ public static Object getStub(final Object type, final String value) {
...
+                throw new GsonBuildException("No matching methods for class " + c.getName());
+            } catch (InvocationTargetException ex) {
+                ex.printStackTrace();
+                throw new GsonBuildException("Error running method for class " + c.getName());
+            }
+
+            return objectCollection;
+
+        } else{
+            throw new RuntimeException("Unexpected JSON entry");
```
Comment (denis-yuen) [inline cwlavro-tools/src/main/java/io/cwl/avro/CWL.java]: This probably should be handled some other way. Perhaps a null return to indicate that the wrong kind of JSON element was passed or a second exception.

**Pull request merged**